apps/blockchain/lib/blockchain/transaction.ex

```diff
@@ -211,19 +211,49 @@ defmodule Blockchain.Transaction do
 
     {expended_gas, refund} = calculate_gas_usage(tx, remaining_gas, sub_state)
 
+    {account_interface_after_receipt, receipt} =
+      if empty_contract_creation?(tx) &&
```

Comment:
There seems to be a lot of repetition in this `if else` condition, but it seems like the _timing_ of when we perform the `AccountInterface.commit` matters. Is that correct? 

If so, is there a different way we could express this? I have a hard time understanding why we need to commit before we pay and refund gas and clean accounts in the `if` statement and after in the `else` statement.